### PR TITLE
Generate fake data script for load-testing

### PR DESCRIPTION
**Pull request by nathanaelhoun**
## Objectives

The idea with this tool is to populate the database with fake data and see the impact on the app's performance, both server and client side.

This script generate fake data:
- using [`faker.js`](https://marak.github.io/faker.js) and interacting with the server using the REST API (with chai), 
- or by directly inserting data into the database (because we can't insert users throught the API, and inserting molecules was easier with the SQL script).

## Changes

- [x] Use `faker.js` to generate random users. As the seed is always the same, the users will be always the same
- [x] Generate random duels that are finished or in progress
- [x] [Updated developer Wiki](https://github.com/ValFraNath/guacamole/wiki/Developer-setup#load-testing)

## Suspected issues seen when testing:
- The number of victories and defeats seems to not be incremented correctly
- Rendering the Create Duel view takes a lot of time with 300+ users

### CODE REVIEW

File: server/db/fill-fake-data.js [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+import { queryPromise } from "./database.js";
+
+chai.use(chaiHttp);
+dotenv.config();
+
+const NUMBER_OF_USERS = process.argv[2] ?? 300;
+const NUMBER_OF_DUELS = process.argv[3] ?? NUMBER_OF_USERS * 10;
+const SEED = 123;
+
+if (NUMBER_OF_USERS < 2 && NUMBER_OF_DUELS > 0) {
```
Comment (valentinperignon) [inline server/db/fill-fake-data.js:21]: Why we must have more than 0 duels if we have less than 2 users?

File: server/db/fill-fake-data.js [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+async function start() {
+  try {
+    // Sleep waiting that the server starts
+    while (!app.isReady) {
+      await new Promise((r) => setTimeout(r, 1000));
+    }
+
+    console.info(`Filling database with ${NUMBER_OF_USERS} users and ${NUMBER_OF_DUELS} duels.`);
+    console.info(`Using seed ${SEED}`);
+    console.info("");
```
Comment (valentinperignon) [inline server/db/fill-fake-data.js:43]: It's just to create a new line?

File: server/db/fill-fake-data.js [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    const fakeName = (faker.name.firstName().substr(0, 1) + faker.name.lastName())
+      .replace(/\s+/g, "")
+      .normalize("NFD")
+      .replace(/[\u0300-\u036f]/g, "")
+      .substr(0, 8)
+      .toLocaleLowerCase();
+
+    if (usedLogins[fakeName]) {
+      i--;
+      continue;
```
Comment (valentinperignon) [inline server/db/fill-fake-data.js:94]: is this continue necessary?

**Review by valentinperignon — APPROVED** [review]
LOOKSGOODTOME
Just a few comments/questions :)

File: server/db/fill-fake-data.js (hunk 1 above)
Comment (nathanaelhoun) [reply server/db/fill-fake-data.js:21]: I'm not sure to understand your question here. The check is to exit if we try to have more than 0 duel with less than 2 users.

File: server/db/fill-fake-data.js (hunk 2 above)
Comment (nathanaelhoun) [reply server/db/fill-fake-data.js:43]: Yes, it is

File: server/db/fill-fake-data.js (hunk 3 above)
Comment (nathanaelhoun) [reply server/db/fill-fake-data.js:94]: Hum, not really I guess!

Comment (nathanaelhoun) [inline server/db/fill-fake-data.js:94]:
```suggestion
```

File: server/db/fill-fake-data.js (hunk 1 above)
Comment (valentinperignon) [reply server/db/fill-fake-data.js:21]: Oh okay, I didn't understand it like that! It's all good then :)

**Review by FrancoisPog — APPROVED** [review]
Waow so many people on our app !

**Pull request merged**